server/src/main/java/org/opensearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -238,10 +238,20 @@ private void doRecovery(final long recoveryId, final StartRecoveryRequest preExi
                     assert recoveryTarget.sourceNode() != null : "can not do a recovery without a source node";
                     logger.trace("{} preparing shard for peer recovery", recoveryTarget.shardId());
                     indexShard.prepareForIndexRecovery();
-                    final long startingSeqNo = indexShard.recoverLocallyUpToGlobalCheckpoint();
+                    boolean isRecoveringReplicaWithRemoteTxLogEnabledIndex = recoveryTarget.state().getPrimary() == false
+                        && indexShard.isRemoteTranslogEnabledOnPrimary();
+                    final long startingSeqNo = isRecoveringReplicaWithRemoteTxLogEnabledIndex
+                        ? indexShard.fetchStartSeqNoFromLastCommit()
+                        : indexShard.recoverLocallyUpToGlobalCheckpoint();
```

Comment:
Maybe we should just abstract it to a renamed`recoverUptoSeqNo` which should be no-op for remote translog

Reply:
done